main_test.go

```diff
@@ -540,3 +540,51 @@ func TestPrintTrimmedError(t *testing.T) {
 	printTrimmedError(strings.NewReader("line 1\nline 2\nline 3"), output)
 	assert.Equal(t, "line 3\n", output.String())
 }
+
+func TestTokenParamPrecedence(t *testing.T) {
+    // Reset environment
+    os.Clearenv()
+
+    // Set both token-related environment variables
+    os.Setenv("TOKEN", "from-token-env")
+    os.Setenv("PLUGIN_TOKEN", "from-plugin-token-env")
+
+    // Set token CLI argument
+    set := flag.NewFlagSet("prefer-arg-over-any-env", 0)
+    c := cli.NewContext(nil, set, nil)
+    set.String("token", "from-token-arg", "")
+
+    // token CLI argument takes precedence over either environment variable
+    assert.Equal(t, "from-token-arg", c.String("token"))
```

Comment:
@montmanu - there more I think about these tests, the less they make sense.
This would really just tests that the library is working, none of this invokes code any plugin code
(i also still haven't figured out how to trigger env parsing of flags without using the full `cli.App()` thing but that's another problem)
Thoughts?

Reply:
i agree that its not testing much of the plugin's code, but its also a somewhat critical part of the plugin's public API, so a bit worried that from a user's e2e POV, we're not well covered on the token parsing / selection logic..

maybe it would be easier for us to figure out a way to cover that via e2e tests? build the plugin, trigger some downstream ci process that would run that version of the plugin against all versions of drone supported by the plugin?

Reply:
Agreed this is important. Let me try something, i think i have an idea what might work.

Reply:
Ok, I think I found a way to get the tests to do something meaningful. Might actually want to replace some others like `TestCheckParams` with the same App.Run() pattern to test the real plugin-code.